subprojects/docs/src/docs/userguide/dep-man/06-publishing/publishing_gradle_module_metadata.adoc

```diff
@@ -123,6 +123,27 @@ The following rules are enforced:
 
 These rules ensure the quality of the metadata produced, and help confirm that consumption will not be problematic.
 
+[[sub:gmm-reproducible]]
+== Making Gradle Module Metadata reproducible
+
+By default, the Gradle Module Metadata file contains a unique id from the build that generated it.
+This means that the file will always be different.
+
+Users can choose to replace this unique id by something else through the configuration of the `buildIdentifier` in their `publication`:
+
+.Configure the build identifier of a publication
+====
+include::sample[dir="snippets/userguide/publishing/javaLibrary/groovy",files="build.gradle[tags=configure-build-id]"]
+include::sample[dir="snippets/userguide/publishing/javaLibrary/kotlin",files="build.gradle.kts[tags=configure-build-id]"]
+====
+
+With the changes above, the generated Gradle Module Metadata file will always be the same, allowing downstream tasks to consider it up-to-date.
+
+[NOTE]
+====
+While the task generating the metadata is never up-to-date for now, this can allow users to have its output remain identical across executions.
```

Comment:
I think this can be clarified to say that is is effectively up-to-date (just Gradle is not showing it).

> The task generating the module metadata files is currently never marked `UP-TO-DATE` by Gradle due to the way it is implemented. However, if the _build identifier_ is configured to not depend on the build invocation anymore, and everything else in the build is `UP-TO-DATE`, the task is effectively up-to-date (it always produces the same output).

Reply:
Except that's not true either. If you were to make a change to the `publication`, the file could end up different while everything non publication related is up to date. That's why I was less specific and focused on that task only.
Makes sense?

Reply:
Yes that makes sense What about using "no build inputs changed":

> The task generating the module metadata files is currently never marked `UP-TO-DATE` by Gradle due to the way it is implemented. However, if the _build identifier_ is configured to not depend on the build invocation anymore, and no build inputs changed, the task is effectively up-to-date (it always produces the same output).

Reply:
Or "no build inputs nor the build itself"